pkg/deviceshifu/deviceshifumqtt/deviceshifumqtt.go

```diff
@@ -45,14 +47,16 @@ func New(deviceShifuMetadata *deviceshifubase.DeviceShifuMetaData) (*DeviceShifu
 		return nil, err
 	}
 
+	mqttInstructions := CreateMQTTInstructions(&base.DeviceShifuConfig.Instructions)
+
 	if deviceShifuMetadata.KubeConfigPath != deviceshifubase.DeviceKubeconfigDoNotLoadStr {
 		switch protocol := *base.EdgeDevice.Spec.Protocol; protocol {
 		case v1alpha1.ProtocolMQTT:
 			mqttSetting := *base.EdgeDevice.Spec.ProtocolSettings.MQTTSetting
-			var mqttServerAddress string
-			if mqttSetting.MQTTTopic == nil || *mqttSetting.MQTTTopic == "" {
-				return nil, fmt.Errorf("MQTT Topic cannot be empty")
+			if &mqttSetting == nil {
```

Comment:
This won't work if `ProtocolSettings` itself is `nil`
Let's remove getting values from MQTT settings for now since both Topic and Address has been removed in deviceShifuMQTT. We can print out a log and add a TODO if user configured `MQTTServerSecret`